README.md: improve documentation and add examples (#28)
DataDog/orchestrion: README.md

## Getting started

Orchestrion automatically adds instrumentation to your source code. This process saves time that engineers would have to spend manually adding the tracer and integrations by replacing that process with a single command.

There are a couple of ways to add instrumentation to your projects using Orchestrion. Orchestrion will also optionally (`-rm`) remove any instrumentation it has added, so this process is reversible.

### Instrument your code before check-in

The quickest way to instrument your code is to download and run Orchestrion on your development machine, and then check in the instrumented code. The executable(s) built with this instrumented code support tracing.

1. Install Orchestrion.

```sh
go install github.com/datadog/orchestrion@latest
```

2. Run Orchestrion to scan and rewrite the codebase.

```sh
## Go to your project root
cd ~/my_go_project

## Run orchestrion on the root directory. This will recursively instrument everything in the directory and its subdirectories.
orchestrion -w ./

## Build your project like normal. If you get errors here, you may need to tidy your go.mod file (go mod tidy)

> go build ./cmd/exe

Reviewer: nit: the docs team might have some guidelines - I just think it's better to provide generic examples
```suggestion
go build ./path/to/entrypoint
```

![Orchestrion](https://upload.wikimedia.org/wikipedia/commons/5/55/Welteorchestrion1862.jpg)

## What it does

Orchestrion processes Go source code and automatically inserts instrumentation.
Currently this instrumentation will create Datadog APM traces for the instrumented code, but future work will include support for OpenTelemetry as well.

## Getting started

Orchestrion will automatically add instrumentation to your source code. This process saves the time that engineers would have to spend manually adding the tracer and integrations by replacing that process with a single command. There are a couple ways it's possible to add instrumentation to your projects using orchestrion.

> There are a couple ways to use orchestrion
> Orchestrion will also optionally (`-rm`) remove any instrumentation it has added, so this process is reversible.

Reviewer 2: ```suggestion
There are a couple of ways to add instrumentation to your projects using orchestrion. Orchestrion will also optionally (`-rm`) remove any instrumentation it has added, so this process is reversible.
```

> ### Instrument your code before check-in.

Reviewer 2: ```suggestion
### Instrument your code before check-in
```
